lib/matplotlib/image.py

```diff
@@ -1200,7 +1200,11 @@ def pilread(fname):
         except ImportError:
             return None
         with Image.open(fname) as image:
-            return pil_to_array(image)
+            array = pil_to_array(image)
+            # Since the handler for PNGs normalizes the image to values from 0 to 1
+            # this should be done for all other images too as it might lead to
+            # confusion if different value ranges appear for different image types.
+            return array / float(array.max())
```

Comment:
Surely neither of these are correct; they will stretch the image contrast. It should be divided by the maximum possible value of the datatype.